realtime/exceptions.py

```diff
@@ -8,3 +8,15 @@ def __init__(self, func_name: str):
 
     def __str__(self):
         return f"A WS connection has not been established. Ensure you call Socket.connect() before calling Socket.{self.offending_func_name}()"
+
+
+class AuthorizationError(Exception):
+    """
+    Raised when there is an authorization failure for private channels
+    """
+
+    def __init__(self, message: str = "Authorization failed for private channel"):
+        self.message: str = message
```

Comment:
```suggestion
    def __init__(self, message: str = None):
        self.message: str = message or "Authorization failed for private channel"
```

Mutable default argument. IMHO.